WeakAuras/GenericTrigger.lua

```diff
@@ -876,39 +876,61 @@ function WeakAuras.ScanEvents(event, arg1, arg2, ...)
   Private.StopProfileSystem("generictrigger " .. system)
 end
 
----@param event string
----@param unit UnitToken
----@param ... any
-function WeakAuras.ScanUnitEvents(event, unit, ...)
-  Private.StartProfileSystem("generictrigger " .. event .. " " .. unit)
-  local unit_list = loaded_unit_events[unit]
-  if unit_list then
-    local event_list = unit_list[event]
-    if event_list then
-      for id, triggers in pairs(event_list) do
-        Private.StartProfileAura(id);
-        Private.ActivateAuraEnvironment(id);
-        local updateTriggerState = false;
-        for triggernum, data in pairs(triggers) do
-          local delay = GenericTrigger.GetDelay(data)
-          if delay == 0 then
-            local allStates = WeakAuras.GetTriggerStateForTrigger(id, triggernum);
-            if (RunTriggerFunc(allStates, data, id, triggernum, event, unit, ...)) then
-              updateTriggerState = true;
+do
+  local skipTheseUnitsInRaid = {
+    player = true,
+    pet = true,
+    party1 = true,
+    party1pet = true,
+    party2 = true,
+    party2pet = true,
+    party3 = true,
+    party3pet = true,
+    party4 = true,
+    party4pet = true,
+  }
+
+  ---@param event string
+  ---@param unit UnitToken
+  ---@param ... any
+  function WeakAuras.ScanUnitEvents(event, unit, ...)
+    Private.StartProfileSystem("generictrigger " .. event .. " " .. unit)
+    local unit_list = loaded_unit_events[unit]
+    if unit_list then
+      local event_list = unit_list[event]
+      if event_list then
+        for id, triggers in pairs(event_list) do
+          Private.StartProfileAura(id);
+          Private.ActivateAuraEnvironment(id);
+          local updateTriggerState = false;
+          for triggernum, data in pairs(triggers) do
+            local skipThisUnit = false
```

Comment:
I took a look at the code and I think it fits better inside RunTriggerFunc next to the already existing code for data.trigger.unit.